FakeServer/Controllers/DynamicController.cs

```diff
@@ -260,6 +325,22 @@ public async Task<IActionResult> ReplaceItem(string collectionId, [FromRoute][Dy
             if (item == null || _ds.IsItem(collectionId))
                 return BadRequest();
 
+            if (_cachingSettings.ETag.Enabled)
```

Comment:
This could be in ETag-class, like IsPreConditionOk. It takes Request, idField etc. as parameters.

```cs
if (!ETag.IsPreConditionOk(Request, _ds,  _dsSettings.IdField...)
     return new StatusCodeResult(StatusCodes.Status412PreconditionFailed);
```

IsPreConditionOk would be something like this?
```cs					
if (!Request.Headers.ContainsKey(HeaderNames.IfMatch))
  return true;

var result = _ds.GetCollection(collectionId).Find(e => ObjectHelper.CompareFieldValueWithId(e, _dsSettings.IdField, id)).FirstOrDefault();

var content = JsonConvert.SerializeObject(result);

var eTag = GenerateETag(content);

var requestETag = Request.Headers[HeaderNames.IfMatch].FirstOrDefault();

return eTag == requestETag;
```